core/src/decoder.rs

```diff
@@ -823,6 +814,49 @@ impl Decoder {
 	}
 }
 
+/// Decodes old address pre-refactor (https://github.com/paritytech/substrate/pull/7380)
+/// and converts it to a MultiAddress, where "old" here means anything before v0.8.26 or 26/2026/46 on polkadot/kusama/westend respectively.
+fn decode_old_address(data: &[u8], cursor: &mut usize) -> Result<substrate_types::Address, Error> {
```

Comment:
AFAIU there's nothing checks that `cursor` and `cursor + 1`are valid indexes into `data slice` but maybe checked somewhere else?

It may panic then.

Reply:
Yeah,

This is part of the larger issue that the function `decode_single` doesn't do any bounds checking on the cursor at all, and panics if anything goes wrong. I'd rather address that in it's own PR though, since it might be a bit of a large refactor so I opened #35 to track that

Reply:
Ok, cool